lib/msf/core/post/windows/task_scheduler.rb

```diff
@@ -0,0 +1,500 @@
+            [
+              OptEnum.new(
+                'ScheduleType', [
+                  true,
+                  'Schedule frequency for the new created task.',
+                  'ONSTART',
+                  %w[MINUTE HOURLY DAILY WEEKLY MONTHLY ONCE ONSTART ONLOGON ONIDLE]
+                ]
+              ),
+              # This defines the amount of minutes/hours/days/weeks/months,
+              # depending on the ScheduleType value. When ONIDLE type is used,
+              # this represents how many minutes the computer is idle before
+              # the task starts. This value is not used with ONCE, ONSTART and
+              # ONLOGON types
+              OptInt.new(
+                'ScheduleModifier', [
+                  false,
+                  'Schedule frequency modifier to define the amount of \'ScheduleType\'',
+                  1
+                ],
+                conditions: ['ScheduleType', 'in', %w[MINUTE HOURLY DAILY WEEKLY MONTHLY ONIDLE] ]
+              ),
+              # Remote task creation does not seem to work on Windows XP and Windows Server 2003
+              OptString.new(
+                'ScheduleRemoteSystem', [
+                  false,
+                  'The remote system to connect to (prefer FQDN to IP). Not compatible with Windows XP/Server 2003.'
+                ]
+              ),
+              # Use the permissions of this remote user account to schedule the
+              # task remotely. It is recommended to inform the domain or '.\'
+              # if it is a local user (e.g. MYDOMAIN\User1 or .\Administrator).
+              # Also, when `ScheduleRemoteSystem` is set and not
+              # `ScheduleUsername` and `SchedulePassword`, the current user
+              # credentials are used.
+              OptString.new(
+                'ScheduleUsername', [
+                  false,
+                  'User account to schedule the task remotely.'
+                ],
+                conditions: ['ScheduleRemoteSystem', 'nin', [nil, '']]
+              ),
+              OptString.new(
+                'SchedulePassword', [
+                  false,
+                  'Password of the user account set in \'ScheduleUsername\'.'
+                ],
+                conditions: ['ScheduleRemoteSystem', 'nin', [nil, '']]
+              ),
+              OptString.new(
+                'ScheduleRunAs', [
+                  false,
+                  'Execute the task under this user account (default: SYSTEM).',
+                  'SYSTEM'
+                ]
+              ),
+              # Hide the task from "schtasks /query" and Task Scheduler by
+              # deleting the associated Security Descriptor registry value.
+              # Note that SYSTEM privileges are needed for this. It will try to
+              # elevate privileges if the session is not already running under
+              # the SYSTEM user.',
+              OptBool.new(
+                'ScheduleObfuscateTask', [
```

Comment:
Obfuscation / evasion techniques don't age well. It's very possible that this technique may not work in a few years but a new one that we don't know about will. In order to make this more future proof for that scenario, would you mind switching this to an enum? I'd name the current obfuscation technique whatever you want, and then just have this default to None, "", Automatic, whatever. Boolean options just don't scale / age well.